pytorch_lightning/utilities/argparse_utils.py

```diff
@@ -2,8 +2,6 @@
 
 rank_zero_deprecation("`argparse_utils` package has been renamed to `argparse` since v1.2 and will be removed in v1.4")
 
-from pytorch_lightning.utilities.argparse import *  # noqa: F403 E402 F401
-
 # for backward compatibility with old checkpoints (versions < 1.2.0)
 # that need to be able to unpickle the function from the checkpoint
 from pytorch_lightning.utilities.argparse import _gpus_arg_default  # noqa: E402 F401 # isort: skip
```

Comment:
@carmocca In theory this should be fine because we still have the import below here for BC. 
Unless we remove the file argparse_utils.py we don't need to worry (yet)

Reply:
The problem here is that the deprecation tests have been removed but not the deprecation message

And if we do, this will be forgotten for eternity haha

Is that okay?

Reply:
I think that the best way would be to add a test for legacy checkpoint which covers this use-case
at this moment we have really very simple legacy model to test with...
so I would merge this as it is and raise a priority issue to update legacy checkpoint tests :rabbit:

Reply:
> And if we do, this will be forgotten for eternity haha
> 
> Is that okay?

No, I will never forget the pain it has caused me.


If you want, I can generate one of these checkpoints and we make a tests that loads it? If someone removes the line, we make sure the test breaks?

Reply:
Nah, no need for that. If spending time on it, it should be on how to figure out the monkey patching

Reply:
yes, can you prepare a model which would cover those issue
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/zero_training.py
then we can regenerate all legacy checkpoints

Reply:
all we need to do is add `self.save_hyperparameters() `to the zero_training model and generate the default hparams form `Trainer.parse_argparse_args`. The problem is that `save_hyperparameters()` doesn't exist that long and changed over time so I fear we won't be able to get this working for too old versions of Lightning.

Reply:
we can regenerate just some "recent" checkpoints...
cc: @kaushikb11